### PR TITLE
Made separate function for getTemplates and updated getPrograms.

### PR DESCRIPTION
**Pull request by bledsoef**
Adjusted the templateSelect controller method to utilize getPrograms more and made a function getTemplates toi remove any unnecessary conditionals.

Fixes issue #458

### CODE REVIEW

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
We need to add tests for the new functionality in `getPrograms` and for `getTemplates`. And while we're at it, we should rename them to `getAllowedPrograms` and `getAllowedTemplates` (or better names if you think of better ones).

File: app/controllers/admin/routes.py [hunk 1]
```diff
@@ -44,13 +44,8 @@ def switchUser():
...
     if g.current_user.isCeltsAdmin or g.current_user.isCeltsStudentStaff:
-        allprograms = []
-        if g.current_user.isCeltsStudentStaff:
-            allprograms = getPrograms(g.current_user)
-            visibleTemplates = []
-        else:
-            allprograms = Program.select().order_by(Program.programName)
-            visibleTemplates = EventTemplate.select().where(EventTemplate.isVisible==True).order_by(EventTemplate.name)
+        allprograms = getPrograms(g.current_user)
+        visibleTemplates = getTemplates(g.current_user)
```
Comment (BrianRamsay) [inline app/controllers/admin/routes.py]: This no longer works.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Get in the habit of testing your pages after you make changes. This breaks just clicking on Create Event

File: app/controllers/admin/routes.py (hunk 1 above)
Comment (Boorclark) [reply app/controllers/admin/routes.py]: resolved

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**